libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/DBOpenHelper.java

```diff
@@ -397,15 +397,17 @@ public static boolean smartStoreExists(Context ctx, String dbNamePrefix,
 
 	static class DBHook implements SQLiteDatabaseHook {
 		public void preKey(SQLiteDatabase database) {
-			// Using sqlcipher 3.x default settings
-			// => should open 3.x databases without any migration
-			database.execSQL("PRAGMA cipher_default_compatibility = 3");
 			// Using sqlcipher 2.x kdf iter because 3.x default (64000) and 4.x default (256000) are too slow
 			// => should open 2.x databases without any migration
 			database.execSQL("PRAGMA cipher_default_kdf_iter = 4000");
 		}
 
+		/**
+		 * Need to migrate for SqlCipher 4.x
+		 * @param database db being processed
+		 */
 		public void postKey(SQLiteDatabase database) {
+			database.rawExecSQL("PRAGMA cipher_migrate");
```

Comment:
I wonder if that set the kdf_iter to 256000 (4.x default) - in the past we noticed that a high kdf_iter made opening the database noticeably slower. I'll do some testing.

Reply:
I just did the work for [iOS](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2925) and checked the run time settings and upgrade scenarios. Everything looks fine.